### PR TITLE
Dell XPS 13-9320

### PR DESCRIPTION
**Pull request by FRidh**
###### Things done

- [x] Tested the changes in your own NixOS Configuration
- [x] Tested the changes end-to-end by using your fork of `nixos-hardware` and
      importing it via `<nixos-hardware>` or Flake input

### CODE REVIEW

File: dell/xps/13-9320/webcam.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ config, pkgs, ... }:
+
+{
+  hardware.firmware = let
+    ivsc-firmware = with pkgs; stdenv.mkDerivation rec {
```
Comment (Mic92) [inline dell/xps/13-9320/webcam.nix:5]: We might potentially want to move this to nixpkgs if it is also useful for other hardware.

File: dell/xps/13-9320/webcam.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+      };
+
+      installPhase = ''
+        mkdir -p $out/lib/firmware/vsc/soc_a1_prod
+
+        cp firmware/ivsc_pkg_ovti01a0_0.bin $out/lib/firmware/vsc/soc_a1_prod/ivsc_pkg_ovti01a0_0_a1_prod.bin
+        cp firmware/ivsc_skucfg_ovti01a0_0_1.bin $out/lib/firmware/vsc/soc_a1_prod/ivsc_skucfg_ovti01a0_0_1_a1_prod.bin
+        cp firmware/ivsc_fw.bin $out/lib/firmware/vsc/soc_a1_prod/ivsc_fw_a1_prod.bin
+      '';
```
Comment (Mic92) [inline dell/xps/13-9320/webcam.nix:23]: meta.description and license would be nice to have.

File: dell/xps/13-9320/webcam.nix (hunk 1 above)
Comment (FRidh) [reply dell/xps/13-9320/webcam.nix:5]: ohh, it is in nixpkgs nowadays as well I see.

Comment (Mic92) [reply dell/xps/13-9320/webcam.nix:5]: Right. In that case, we should pull it from there.

Comment (bubuntux) [reply dell/xps/13-9320/webcam.nix:5]: mr created to cover add the necessary symbolic links https://github.com/NixOS/nixpkgs/pull/295508

File: dell/xps/13-9320/default.nix [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{ lib, pkgs, ... }: {
+  imports = [
+    ../../../common/cpu/intel
+    ../../../common/pc/laptop
+    ../../../common/pc/ssd
+    ./webcam.nix
```
Comment (bubuntux) [inline dell/xps/13-9320/default.nix:6]: after these commits [1](https://github.com/NixOS/nixpkgs/pull/295508) and [2](https://github.com/NixOS/nixpkgs/pull/296164) , we probably can remove the webcam file and just add 

```
 hardware.ipu6 = {
    enable = true;
    platform = "ipu6ep";
  };
```

**Comment by Mic92** [comment]
The pull request seems to be abandoned. Please re-open if you want to continue working on it.

**Pull request closed without merging**